apps/req.c

```diff
@@ -517,6 +517,14 @@ int req_main(int argc, char **argv)
             goto end;
         }
 
+        if ((newkey > OPENSSL_RSA_MAX_MODULUS_BITS && pkey_type == EVP_PKEY_RSA)
+            || (newkey > OPENSSL_DSA_MAX_MODULUS_BITS && pkey_type == EVP_PKEY_DSA))
+               BIO_printf(bio_err,
+                          "# Warning: It is not recommended to use more than %d bit for RSA keys #\n"
+                          "#          and more than %d bit for DSA keys. Your key size is %ld! #\n"
+                          "#          Larger key size may behave not as expected.                   #\n",
+                          OPENSSL_RSA_MAX_MODULUS_BITS, OPENSSL_DSA_MAX_MODULUS_BITS, newkey);
```

Comment:
Ok, I will do that.

My intend was that a user has the alternative to switch from DSA 16384 (warning) to RSA 16384 (ok).

Additional to your draft it would be possible add a hint if OPENSSL_DSA_MAX_MODULUS_BITS < userKeySie <= OPENSSL_RSA_MAX_MODULUS_BITS is fulfilled.

I think your draft is the best solution in the way of simple code and minium of text.
So I will take it.

Reply:
See 719feec

See also 94f8a66 (may thrown)